evap/staff/templates/staff_semester_view_evaluation.html

```diff
@@ -188,6 +188,56 @@
             <a href="{% url 'staff:evaluation_copy' evaluation.id %}" class="btn btn-sm btn-light" data-bs-toggle="tooltip" title="{% translate "Copy" %}">
                 <span class="fas fa-copy"></span>
             </a>
+            {% if not evaluation.has_exam %}
+                <confirmation-modal type="submit" id="exam_creation_modal" name="evaluation_id" value="{{ evaluation.id }}" confirm-button-class="btn-primary" form="exam_creation_form_{{ evaluation.id }}">
+                    <span slot="title">{% translate 'Create exam evaluation' %}</span>
+                    <span slot="action-text" id="submit_button">{% translate 'Create exam evaluation' %}</span>
+                    <span slot="question">
+                        {% blocktranslate trimmed %}
+                            Create an exam evaluation based on this evaluation. This will copy all the participants and contributors from the original evaluation. It will set the weight of the original evaluation to 9 and its end date will be set to the day before the exam.
+                        {% endblocktranslate %}
+                    </span>
+                    <div slot="extra-inputs">
+                        <label for="exam_date">{% translate 'Exam Date:' %}</label>
+                        <input type="date" id="exam_date" name="exam_date" data-must-before="{{ evaluation.vote_start_datetime }}" class="form-control" required form="exam_creation_form_{{ evaluation.id }}"/>
+                    </div>
+
+                    <button slot="show-button" type="button" class="btn btn-sm btn-light" title="{% translate 'Create exam evaluation' %}" data-bs-placement="top" data-bs-toggle="tooltip"><span class="fas fa-file-pen fa-fw"></span></button>
+
+                    <script type="text/javascript">
+                        document.getElementById('submit_button').addEventListener('click', function(event) {
+                            var inputDate = new Date(document.getElementById('exam_date').value);
+                            var evaluationDate = new Date(document.getElementById('exam_date').dataset.mustBefore);
```

Comment:
Maybe you have already considered this, but I would think that this validation usually happens in a [`change` event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) handler on the element itself. So something like

```javascript
document.querySelectorAll("input[type=date][data-must-before]").forEach(input => {
    input.addEventListener("change", () => {
        // check stuff with input.value and input.dataset.mustBefore
        // and call setCustomValidity
    });
});
```

This code should then be placed outside the Django `{% for ... %}` loop so it is not duplicated - the `additional_javascript` block seems appropriate